src/modules/Modal/Modal.js

```diff
@@ -1,48 +1,69 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
+import React, { PropTypes } from 'react'
+import cx from 'classnames'
 import META from '../../utils/Meta'
-import ModalContent from './ModalContent'
 import ModalHeader from './ModalHeader'
-import ModalFooter from './ModalFooter'
-
-export default class Modal extends Component {
-  static propTypes = {
-    children: PropTypes.any,
-    className: PropTypes.string,
-    settings: PropTypes.object,
-  }
-
-  state = { isShown: false }
-
-  static Content = ModalContent
-  static Header = ModalHeader
-  static Footer = ModalFooter
-
-  showModal = () => {
-    this.setState({ isShown: true })
-  }
-
-  hideModal = () => {
-    this.setState({ isShown: false })
-  }
-
-  static _meta = {
-    name: 'Modal',
-    type: META.type.module,
-  }
-
-  render() {
-    const classes = classNames(
-      'ui',
-      this.props.className,
-      'modal',
-      { 'transition visible active': this.state.isShown },
-    )
-
-    return (
-      <div {...this.props} className={classes}>
-        {this.props.children}
-      </div>
-    )
-  }
+import ModalContent from './ModalContent'
+import ModalActions from './ModalActions'
+import {
+  getUnhandledProps,
+  useKeyOnly,
+} from '../../utils/propUtils'
+
+function Modal(props) {
+  const {
+    children, className,
+    active, basic, size, long,
+  } = props
+
+  const classes = cx('ui',
+    useKeyOnly(active, 'transition visible active'),
+    useKeyOnly(basic, 'basic'),
+    useKeyOnly(long, 'long'),
+    size,
+    'modal',
+    className,
+  )
+
+  const rest = getUnhandledProps(Modal, props)
+
+  return (
+    <div className={classes} {...rest} >
+      {children}
+    </div>
+  )
+}
+
+Modal._meta = {
+  name: 'Modal',
+  type: META.type.module,
+  props: {
+    size: ['fullscreen', 'large', 'small'],
+  },
 }
+
+Modal.propTypes = {
+  /** Primary content of the modal. Consider using ModalHeader, ModalContent or ModalActions here */
+  children: PropTypes.any,
+
+  /** Classes to add to the modal className */
+  className: PropTypes.string,
+
+  /** An active modal is visible on the page */
+  active: PropTypes.bool,
+
+  /** A modal can reduce its complexity */
+  basic: PropTypes.bool,
+
+  /** A modal can vary in size */
+  size: PropTypes.oneOf(Modal._meta.props.size),
+
+  /** When your modal content exceeds the height of the browser the scrollable area will automatically
+   * expand to include just enough space for scrolling, without scrolling the page below. */
+  long: PropTypes.bool,
```

Comment:
This prop is being used to add a className, however, there is no `long` className supported in SUI.  The `long` class is shown in their docs only as an example.  It does not actually do anything.

For now, let's remove the `long` prop.